### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by gnufied**
Adds e2e tests for local-storage-operaror. Currently I am not validating local PV creation. Those can be added as a follow up.

### CODE REVIEW

**Comment by gnufied** [comment]
It works

```
 sed -i s,quay.io/openshift/origin-local-storage-operator:latest,quay.io/openshift/origin-local-storage-operator:latest, /tmp/tmp.Gamz0WQuKq
++ pwd
+ TEST_NAMESPACE=default
+ go test ./test/e2e/... -root=/home/shared/go_projects/src/github.com/openshift/local-storage-operator -kubeconfig=/home/hekumar/.kube/config -globalMan /tmp/tmp.y1sM4QQ5Mo -namespacedMan /tmp/tmp.Gamz0WQuKq -v -parallel=1 -singleNamespace
=== RUN   TestLocalStorageOperator

--- PASS: TestLocalStorageOperator (22.49s)
    localstorage_test.go:94: Initializing cluster resources...
    client.go:57: resource type ServiceAccount with namespace/name (default/local-storage-operator) created
    client.go:57: resource type Role with namespace/name (default/local-storage-operator) created
    client.go:57: resource type RoleBinding with namespace/name (default/local-storage-operator) created
    client.go:57: resource type ClusterRole with namespace/name (default/local-storage-operator) created
    client.go:57: resource type ClusterRoleBinding with namespace/name (default/local-storage-operator) created
    client.go:57: resource type ClusterRoleBinding with namespace/name (default/local-operator-pv-binding) created
    client.go:57: resource type Deployment with namespace/name (default/local-storage-operator) created
    localstorage_test.go:99: Initialized cluster resources
    localstorage_test.go:104: Found namespace: default
    localstorage_test.go:109: Waiting for local-storage-operator to be ready...
    wait_util.go:51: Deployment available (1/1)
    client.go:57: resource type LocalVolume with namespace/name (default/test-local-disk) created
    localstorage_test.go:141: Daemonset available (3/3)
    localstorage_test.go:141: Daemonset available (3/3)
    client.go:75: resource type LocalVolume with namespace/name (default/test-local-disk) successfully deleted
    client.go:75: resource type Deployment with namespace/name (default/local-storage-operator) successfully deleted
    client.go:75: resource type ClusterRoleBinding with namespace/name (default/local-operator-pv-binding) successfully deleted
    client.go:75: resource type ClusterRoleBinding with namespace/name (default/local-storage-operator) successfully deleted
    client.go:75: resource type ClusterRole with namespace/name (default/local-storage-operator) successfully deleted
    client.go:75: resource type RoleBinding with namespace/name (default/local-storage-operator) successfully deleted
    client.go:75: resource type Role with namespace/name (default/local-storage-operator) successfully deleted
    client.go:75: resource type ServiceAccount with namespace/name (default/local-storage-operator) successfully deleted
PASS
ok      github.com/openshift/local-storage-operator/test/e2e    27.926s

```

**Comment by gnufied** [comment]
cc @j-griffith

File: hack/test-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  set +e
+
+  cat $manifest > $ARTIFACT_DIR/manifest
+  cat $global_manifest > $ARTIFACT_DIR/global_manifest
+
+  exit $return_code
+}
+trap cleanup exit
+
+if [ -n "${IMAGE_FORMAT:-}" ] ; then
```
Comment (jsafrane) [inline hack/test-e2e.sh]: why `${xxx:-}`? Is it the same as plain `"${xxx}"`?

File: hack/test-e2e.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+ARTIFACT_DIR=${ARTIFACT_DIR:-_output}
+manifest=$(mktemp)
+global_manifest=$(mktemp)
+
+cleanup(){
+  local return_code="$?"
+  set +e
+
+  cat $manifest > $ARTIFACT_DIR/manifest
```
Comment (jsafrane) [inline hack/test-e2e.sh]: * a comment that this is saving manifest in the test results for debugging would be useful.
* nit: `cp` would perhaps look better.
* `rm -f` of the temp files would be great in the end.

File: hack/test-e2e.sh (hunk 1 above)
Comment (gnufied) [reply hack/test-e2e.sh]: fixed.

File: hack/test-e2e.sh (hunk 2 above)
Comment (gnufied) [reply hack/test-e2e.sh]: fixed

**Comment by gnufied** [comment]
@jsafrane addressed the review comments. PTAL

**Pull request merged**